File: docs/docs/examples/scripts/TestAction.py

# Creates a frame that contains a text area. Prints value 
# of variable named context in the text area.

from javax import swing
from java import awt
from java import lang

# Create frame with border layout
frame = swing.JFrame("TestAction.py")
frame.getContentPane().layout = awt.BorderLayout()

# Add text field
textField = swing.JTextArea()
frame.getContentPane().add(textField,awt.BorderLayout().CENTER)

# Size and show frame 
frame.setBounds(100,100,300,200);
frame.setVisible(1);

# Print context in text area 
try: 
   textField.append("Context = "+context+"\n")
except:
   textField.append("Context not defined\n")

File: website/docs/examples/scripts/TestAction.java

import java.awt.*;
import javax.swing.*;

import org.relayirc.swingui.*;

public class TestAction implements IRelayRunnable {

   public void run( Object context ) {

      JFrame frame = new JFrame();
      frame.setSize(200,200);
      frame.setVisible(true);

      frame.getContentPane().setLayout( new BorderLayout() );

      JTextArea tarea = new JTextArea();
      frame.getContentPane().add( tarea, BorderLayout.CENTER );

      tarea.append("TestAction");
      tarea.append("Context = "+context);
   }

}

